chainntnfs/tx_notifier.go

```diff
@@ -1115,9 +1117,41 @@ func (n *TxNotifier) DisconnectTip(blockHeight uint32) error {
 		}
 	}
 
-	// Finally, we can remove the transactions we're currently watching that
-	// were included in this block height.
+	// We'll also go through our watched outpoints and attempt to drain
+	// their dispatched notifications to ensure dispatching notifications to
+	// clients later on is always non-blocking.
+	for spendHeight, ops := range n.opsBySpendHeight {
+		// We're only interesting in outpoints whose spending
+		// transaction was included at the height being disconnected.
+		if spendHeight != blockHeight {
```

Comment:
just index `n.opsBySpendHeight[blockHeight]` instead of looping?

Reply:
Indeed - don't know what I was thinking. Fixed!